src/index.ts

```diff
@@ -37,10 +52,178 @@ function gistIdFromTwiddleUrl(twiddleUrl: string): string {
+/**
+ * Normalize the generated twiddle to a ember-like structure.
+ * @param targetDir
+ */
+function normalizeTwiddle(targetDir: string) {
+  const appDir = path.join(targetDir, 'app');
+  generateDir(appDir);
+  const files = fs.readdirSync(targetDir);
+  for (let i in files) {
+    const file = files[i];
+    const dirPath = file.split('.');
+    const fileName = dirPath.splice(-2, 2);
+    const newPath = path.join(appDir, ...dirPath);
+    const filePath = path.join(newPath, fileName.join('.'));
+    // If the file parts length is more than 2 then there is a directory structure expected.
+    if (file.split('.').length > 2) {
+      generateDir(path.join(appDir, ...dirPath));
+    }
+    if (fileName.length === 2 && fs.existsSync(newPath)) {
+      try {
+        if (fs.existsSync(filePath)) {
+          continue;
+        } else {
+          fs.renameSync(path.join(targetDir, file), filePath);
+        }
+      } catch (err) {
+        throw err;
+      }
+    }
+  }
+}
+
+async function getLatestEmberVersion(verString: string) {
+  const response = await fetch('https://api.github.com/repos/ember-cli/ember-new-output/tags?per_page=300');
+  const responseJson = await response.json();
+  const ver = semver.parse(verString);
+  const verMap = new Map<semver.SemVer, GitTag>(responseJson.map((tag: GitTag) => [semver.parse(tag.name), tag]));
+  if (ver) {
+    const bestVer = semver.maxSatisfying([...verMap.keys()], `${ver.major}.${ver.minor}`);
+    if (bestVer) {
+      return verMap.get(bestVer);
+    }
+  }
+  throw new Error('The version does not exist');
+}
+/**
+ * Augment the already generated ember-like twiddle project app into a real ember app
+ * by augmenting the real Ember app's structure on top of the twiddle app.
+ * @param twiddleData
+ * @param projectName
+ */
+async function augmentEmberApp(twiddleData: any, projectName: string) {
+  const twiddleItem = JSON.parse(JSON.stringify(twiddleData));
+  const latestEmberVersion = await getLatestEmberVersion(twiddleItem.dependencies.ember);
+  if (latestEmberVersion) {
+    const emberAppDir = await unzipAndExtractFromUrl(
+      `https://${GIT_DOMAIN}/ember-cli/ember-new-output`,
+      `archive/${latestEmberVersion.name}.zip`,
+      path.join('.', projectName)
+    );
+    if (emberAppDir) {
+      const packageData = getJsonFileData(emberAppDir, 'package');
+      const packageDevDep = packageData.devDependencies;
+      const twiddleAddons = twiddleData.addons;
+      let isDevDependencyInTwiddle = false;
+      Object.keys(twiddleAddons).forEach(item => {
+        if (!packageDevDep.hasOwnProperty(item)) {
+          isDevDependencyInTwiddle = true;
+          packageDevDep[item] = twiddleAddons[item];
+        }
+      });
+      // If there are dev dependencies present in twiddle, add them to the package json of the newly generated ember app.
+      if (isDevDependencyInTwiddle) {
+        fs.writeFileSync(path.join(emberAppDir, 'package.json'), JSON.stringify(packageData));
```

Comment:
fs-extra has a [`writeJSONSync`](http://adilapapaya.com/docs/fs-extra/#writejsonfileobjectoptionscallback) that's good for this